integrations/astro-code-snippets.ts

```diff
@@ -0,0 +1,259 @@
+import type { AstroIntegration } from 'astro';
+import type { BlockContent, Root, Parent } from 'mdast';
+import type { Plugin, Transformer } from 'unified';
+import type { BuildVisitor } from 'unist-util-visit/complex-types';
+import rangeParser from 'parse-numeric-range';
+import { visit } from 'unist-util-visit';
+
+const CodeSnippetTagname = 'AutoImportedCodeSnippet';
+const LanguageGroups = {
+	code: ['astro', 'cjs', 'htm', 'html', 'js', 'jsx', 'mjs', 'svelte', 'ts', 'tsx', 'vue'],
+	data: ['env', 'json', 'yaml', 'yml'],
+	styles: ['css', 'less', 'sass', 'scss', 'styl', 'stylus'],
+	textContent: ['markdown', 'md', 'mdx'],
+};
```

Comment:
I actually had a version that does not use hardcoded extensions before, but it caused various false positives, e.g. interpreting valid semver version numbers like `1.4.x` or `1.0.0-feature.rc` as file names. The language groups also help to avoid incorrectly considering our "edge case" comment to be a valid filename because `css` and `html` belong to different groups:

````md
```html
<head>
  <!-- Local: /public/styles/global.css -->
  <link rel="stylesheet" href="/styles/global.css" />
  <!-- External -->
  ...
</head>
```
````

I'm happy to take care of any updating that may be required for these lists. :)